portfolio/src/main/webapp/js/comments.js

```diff
@@ -1,9 +1,26 @@
+let commentAuthData = null;
 /**
  * Initializes the comment system.
- *
- * Adds a submission event listener to the comment form and loads the comments.
  */
 function initCommentsSystem() {
+  getAuthorization().then(() => {
+    applyAuthorizationToUI();
+    loadComments();
+  }).catch((err) => {
+    console.error(err);
```

Comment:
Avoid global variables if possible. Take a look at https://stackoverflow.com/questions/484635/are-global-variables-bad, the top-rated answer does a real good job of explaining why global variables are usually bad.

Rather than reading/writing a global variable, your functions should just return a value to be used by a different function, or take in the variable as an argument. In this case you would change `getAuthorization()` to return `commentAuthData` and pass that to `applyAuthorizationToUI`. it would look something like this:
```suggestion
  getAuthorization().then((commentAuthData) => {
    applyAuthorizationToUI(commentAuthData);
    loadComments();
  }).catch((err) => {
    console.error(err);
```

Reply:
Thanks for the resource, that does make sense, and every time I do use a global variable I tend to cringe a bit so I'm happy my initial reactions are correct haha. 

However, I am making the authorization data global so that I can reference it from within various parts of the comments system. Maybe this is bad design? For example I intend to use it within the `loadComments()` function to determine which comments should have a delete button. It would be easy enough to make it an argument in the above example, but loadComments is also called from the refresh button. Thoughts?

Reply:
Also another thing I should make note of here is to not make certain design decisions based on future intended changes. PR reviewers can't read my mind lol.

Reply:
Yes I think used correctly, TODOs are great for this. Nothing is perfect, and when I find TODOs that indicate where maybe shortcuts were taken or things can be improved, that helps a lot more than having to guess what the author was thinking at the time. 

From a C++ style guide (but it's in like every style guide):
<img width="841" alt="Untitled" src="https://user-images.githubusercontent.com/2067277/84222011-6337d180-aa8b-11ea-8eae-4d0ac442bb83.png">